lua/quicker/treesitter.lua

```diff
@@ -0,0 +1,91 @@
+---@alias quicker.LangRegions table<string, number[][][]>
+
+local M = {}
+
+M.cache = {} ---@type table<number, table<string,{parser: vim.treesitter.LanguageTree, highlighter:vim.treesitter.highlighter, enabled:boolean}>>
+local ns = vim.api.nvim_create_namespace("quicker.treesitter")
+
+local TSHighlighter = vim.treesitter.highlighter
+
+local function wrap(name)
+  return function(_, win, buf, ...)
+    if not M.cache[buf] then
+      return false
+    end
+    for _, hl in pairs(M.cache[buf] or {}) do
+      if hl.enabled then
+        TSHighlighter.active[buf] = hl.highlighter
+        TSHighlighter[name](_, win, buf, ...)
+      end
+    end
+    TSHighlighter.active[buf] = nil
+  end
+end
+
+M.did_setup = false
+function M.setup()
+  if M.did_setup then
+    return
+  end
+  M.did_setup = true
+
+  vim.api.nvim_set_decoration_provider(ns, {
+    on_win = wrap("_on_win"),
+    on_line = wrap("_on_line"),
+  })
+
+  vim.api.nvim_create_autocmd("QuickFixCmdPost", {
+    group = vim.api.nvim_create_augroup("quicker.treesitter.hl", { clear = true }),
+    callback = function(ev)
+      M.cache[ev.buf] = nil
+    end,
+  })
```

Comment:
What problem is this solving? I don't rely on `QuickFixCmdPost` in the rest of this plugin because some methods of setting/updating the quickfix do not trigger it.